components/molecule/modal/src/index.js

```diff
@@ -0,0 +1,216 @@
+    document.addEventListener('keydown', this._onKeyDown)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keydown', this._onKeyDown)
+  }
+
+  _onKeyDown = event => {
+    if (this.state.open === false || this.props.closeOnEscKeyDown === false)
+      return
+    if (SUPPORTED_KEYS.includes(event.key)) {
+      this._closeModal()
+      event.preventDefault()
+    }
+  }
+
+  _preventScrollIfNeeded = e => {
+    if (this.noScroll) e.preventDefault()
+  }
+
+  _avoidOverscroll = () => {
+    const {
+      clientHeight,
+      offsetHeight,
+      scrollTop,
+      scrollHeight
+    } = this.contentDOMEl
+    const currentScroll = scrollTop + offsetHeight
+    // check if the content has to scroll in order to prevent the default
+    // behaviour of the touchmove in case we don't need the scroll
+    // that fixes the weird overscroll on ios and android
+    this.noScroll = scrollHeight <= clientHeight
+
+    if (scrollTop === 0) {
+      this.contentDOMEl.scrollTop = 1
+    } else if (currentScroll >= scrollHeight) {
+      this.contentDOMEl.scrollTop = scrollTop - 1
+    }
+  }
+
+  _closeModal = () => {
+    this._toggleWindowScroll(false)
+    this.setState({open: false})
+    this.props.onClose()
+  }
+
+  _toggleWindowScroll(disableScroll) {
+    window.document.body.classList.toggle('is-modal-open', disableScroll)
+  }
+
+  _handleCloseClick = () => {
+    this._closeModal()
+  }
+
+  _handleOutsideClick = event => {
+    if (this.props.closeOnOutsideClick && event.target === this.wrapperDOMEl) {
+      this._closeModal()
+    }
+  }
+
+  _renderCloseIcon = () => {
+    const {iconClose} = this.props
+    return (
+      <button
+        type="button"
+        className={suitClass({element: 'close'})}
+        onClick={this._handleCloseClick}
+      >
+        {iconClose}
+      </button>
+    )
+  }
+
+  _renderHeader = () => {
+    const {header} = this.props
+    return (
+      <div
+        className={suitClass({element: 'header'})}
+        onTouchMove={e => e.preventDefault()}
+      >
+        {header}
+        {this._renderCloseIcon()}
+      </div>
+    )
+  }
+
+  _renderNoHeader = () => {
+    return (
+      <div className={suitClass({element: 'no-header'})}>
+        {this._renderCloseIcon()}
+      </div>
+    )
+  }
+
+  componentWillReceiveProps({open, disableWindowScroll}) {
+    if (open && disableWindowScroll) {
+      this._toggleWindowScroll(true)
+    }
+
+    if (open !== this.state.open) {
+      this.setState({open})
+    }
+  }
+
+  render() {
+    const {header, children} = this.props
+
+    const wrapperClassName = cx(suitClass({}), {
+      'is-open': this.state.open,
+      [suitClass({modifier: 'verticallyCentered'})]: this.props.centerVertically
+    })
+
+    const dialogClassName = cx(suitClass({element: 'dialog'}), {
+      [suitClass({element: 'dialog--full'})]: this.props.fitWindow
+    })
+
+    return (
+      <div
+        className={wrapperClassName}
+        ref={node => {
```

Comment:
please use the React.createRef() API as callbacks ref was used by older react versions